### ISSUE TITLE
Superscript/subscript support

### ISSUE THREAD

**Issue by palemieux** [w3c/imsc#583]
IMSC currently does not support superscript/subscript text, which is desirable (if not essential) in some languages.

French, for example, uses superscript to [abbreviate ordinal numerals](https://www.academie-francaise.fr/abreviations-des-adjectifs-numeraux). The same is true in English, albeit perhaps less necessary, e.g., _1st_ might be as acceptable as _1<sup>st</sup>_.

![image](https://github.com/user-attachments/assets/5cf3bdfa-f132-4e4d-bafc-f1ab1f2adb9c)

Couple of observations:

* Unicode does not have superscripts characters for all ordinal numerals, e.g., "er", "re", etc.
* TTML2 does not support HTML `sup`/`sub` markup.
* TTML2 supports `tts:fontVariant="super"` which is mapped to CSS `font-variant-position`, but (a) few fonts have super variants and (b) subscripting/superscripting seems more of a semantic or layout issue than a font/glyph selection issue.

CEA 708 has the concept of superscript/subscript (offset pen attribute).

In fact it looks like `tts:fontVariant` was [introduced, at least in part, to match CEA 708 capabilities](https://www.w3.org/AudioVideo/TT/tracker/issues/233), although, as mentioned above, it is not clear that it is a perfect match (or at least the mapping to CSS is not ideal).

**Comment by skynavga** [comment]
@nigelmegitt @palemieux Note that [#derivation-fontVariant](https://www.w3.org/TR/2021/CR-ttml2-20210309/#derivation-fontVariant) says that [`font-variation-position`]() applies for `normal`, `super`, and `sub`, which in turn says [here](https://www.w3.org/TR/2018/REC-css-fonts-3-20180920/#propdef-font-variant-position) that:

>Because of the semantic nature of subscripts and superscripts, when the value is either ‘[sub](https://www.w3.org/TR/2018/REC-css-fonts-3-20180920/#sub)’ or ‘[super](https://www.w3.org/TR/2018/REC-css-fonts-3-20180920/#super)’ for a given contiguous run of text, if a variant glyph is not available for all the characters in the run, simulated glyphs should be synthesized for all characters using reduced forms of the glyphs that would be used without this feature applied. This is done per run to avoid a mixture of variant glyphs and synthesized ones that would not align correctly. In the case of OpenType fonts that lack subscript or superscript glyphs for a given character, user agents must synthesize appropriate subscript and superscript glyphs.

**Comment by palemieux** [comment]
@skynavga in your experience/mind, subscript/superscript is no different than bold and italic, i.e., a different variation of a font for a given character?

**Comment by palemieux** [comment]
I was also surprised that Chrome did not support `font-variation-position: super` if it is equivalent to `<sup>`

https://codepen.io/palemieux/pen/gONjEVx

**Comment by skynavga** [comment]
@palemieux 

>in your experience/mind, subscript/superscript is no different than bold and italic, i.e., a different variation of a font for a given character?

Yes, for higher end fonts (where the designer pays attention to these matters). However, many fonts contain a small set of sub/super glyphs without having the corresponding OpenType substitution table, and, indeed, Unicode itself codes a few sub/sup variants as characters in their own right. I don't actually recall seeing an independent font resource for sup or sub or both (in the sense that one encounters bespoke bold, italic, and bold-italic font resources).

A good text layout/rendering engine will take heed of the language I cited from CSS3 Fonts regarding the need to synthesize glyphs on demand. Indeed, I used this approach to synthesize ruby glyphs in [`ttt/ttt-ttpe`](https://github.com/skynav/ttt/tree/master/ttt-ttpe).

**Comment by palemieux** [comment]
@skynavga Any reason why `font-variation-position: super` is not supported in Chrome, but `<sup>` is? Could we say, in TTML2, that `tts:fontVariant="super"` is the same as `<sup>`?

**Comment by skynavga** [comment]
> @skynavga Any reason why `font-variation-position: super` is not supported in Chrome, but `<sup>` is? Could we say, in TTML2, that `tts:fontVariant="super"` is the same as `<sup>`?

Couldn't say. Haven't been connected to the Chromium project in ages (is it still a thing?). I wouldn't want TTML2 to make a reference to HTML5 to obtain the semantics of `<sup>`. I think the semantics described in CSS3 Fonts are just fine, as it gives plenty of latitude for implementation behavior. Note, however, that TTML2 doesn't normatively use the CSS3 Fonts semantics, but, rather, it is merely an indirect reference as such. Think of it as a hint.

**Comment by skynavga** [comment]
Ok. I view this as an implementation matter. TTML2 doesn't actually reference `font-variant-position` in a normative manner. So it is up to a presentation processor's implementation to decide how to render `tts:fontVariant`, whether it be custom rendering code, mapping to `font-variant-position`, `font-size` with `vertical-align`, or `<sup>`, etc.

Notwithstanding this, I agree that the current normative language defining `tts:fontVariant` focuses only on glyph selection without mentioning fallback strategies in the absence of selectable glyphs. I wouldn't object to a Note being added (to §10.2.23) that elaborates potential fallbacks, which I would probably write in terms similar to that found in CSS3 Fonts as cited above).

**Comment by nigelmegitt** [comment]
For the use case where both a superscript and subscript are needed (chemical symbols etc), vertically above each other, I think `font-variant-position` does not work, but Firefox at least does seem to do what CSS says and synthesise superscript and subscript for ordinary letter-like symbols.

I created a [codepen](https://codepen.io/nigelmegitt/pen/VwJRYyR) to demonstrate different HTML and CSS techniques for superscript and subscript. In many ways the easiest to implement is the one using table layout, though I haven't worked out a more elegant way to get only superscripts to work than putting a non-breaking space character into a `<span>` so that there's something occupying the subscript row, which pushes the superscript row up to the top.

**Comment by nigelmegitt** [comment]
Potentially relevant data point: https://github.com/WebKit/standards-positions/issues/525#issuecomment-3288611736

**Comment by himorin** [comment]
specific comment on this feature from a11y review made at https://github.com/w3c/a11y-request/issues/116#issuecomment-3298869859

(please file another tracker if anyone feels appropriate....)
